### ISSUE TITLE
Missing support for async iterables and iterators

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#447]
**Describe the bug**
pyright does not recognise `AsyncIterable` and `AsyncIterator` as valid return types of asynchronous generator functions mirroring synchronous generator annotations:

```
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Expression of type 'Iterator[int]' cannot be assigned to yield type 'AsyncIterator[int]'\n  '__anext__' is not present\n  '__aiter__' is not present",
	"source": "pyright",
	"startLineNumber": 6,
	"startColumn": 15,
	"endLineNumber": 6,
	"endColumn": 21
}
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "'Awaitable[AsyncIterator[int]]' is not iterable\n  '__aiter__' method not defined",
	"source": "pyright",
	"startLineNumber": 10,
	"startColumn": 30,
	"endLineNumber": 10,
	"endColumn": 35
}
```

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `AsyncIterator` would be a valid return type.

**Screenshots or Code**
```py
from typing import AsyncIterator


async def foo() -> AsyncIterator[int]:
    for number in [1, 2, 3]:
        yield number


async def bar():
    return [f async for f in foo()]
```

**VS Code extension or command-line**
extension, 1.1.11

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next published version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.1.12, which I just published.

**Comment by layday** [comment]
I believe this should also apply to `AsyncIterable`s, not just `AsyncIterator`s - sorry if I wasn't clear in my report :(

**Comment by erictraut** [comment]
Ah yes, you're correct. The Python documentation specifically states that either `AsyncIterable` or`AsyncIterator` can be used.

**Comment by erictraut** [comment]
This will be fixed in the next version.

**Comment by erictraut** [comment]
This is now fixed in 1.1.13, which I just published.
